scripts/add_electricity.py

```diff
@@ -801,6 +802,55 @@ def estimate_renewable_capacities_irena(
             ] * float(p_nom_max)
 
 
+def attach_enhanced_geothermal(n):
+
+    egs_potential = pd.read_csv(snakemake.input["egs_potentials"], index_col=[0, 1])
+    assert egs_potential.index.names == ['network_region', 'capital_cost[$/kW]']
+
+    idx = pd.IndexSlice
+
+    n.add(
+        "Bus",
+        "EGS",
+        carrier="geothermal heat",
+        unit="MWh_th",
+    )
+
+    n.add(
+        "Generator",
+        "EGS",
+        bus="EGS",
+        carrier="geothermal heat",
+        p_nom_extendable=True,
+    )
+
+    for bus in tqdm(
+        egs_potential.index.get_level_values(0).unique(),
+        desc="Adding enhanced geothermal",
+        ):
+
+        ss = egs_potential.loc[idx[bus, :]]
+
+        # Loop over supply‐curve point (i.e. each potential) for this region
+        for i, (capital_cost, row) in enumerate(ss.iterrows()):
+            capacity = row["available_capacity[MW]"]
+            # Convert the cost index from $/kW to $/MW and annuitize the capex
+            capital_cost = float(capital_cost) * 1000 # $/kW -> $/MW
+
+            identifier = f"{bus}_curve{i}"
+
+            n.add(
+                "Link",
+                name=f"EGS electricity {identifier}",
+                bus0="EGS",
+                bus1=bus,
+                carrier="enhanced geothermal ORC",
+                p_nom_max=capacity,
+                capital_cost=capital_cost,
+                p_nom_extendable=True,
+            )
```

Comment:
As a technical comment, this representation is perfectly aligned with what we are using in the sector-coupled part. In the power model, `Links` represent DC connections and are treated accordingly to the workflow (e.g. it's expected to have `dc` and `underground_fraction` fields). Not sure about implications for the solution. My guess is that it should be alright if we are interested in sector-coupled runs only. But it's worth double-checking